site/app/libraries/database/DatabaseQueries.php

```diff
@@ -112,18 +112,32 @@ public function loadThreads($announcements){
         return $this->course_db->rows();
     }
 
+    public function createReply($user, $content, $thread_id, $anonymous, $type, $parent_post, $hasAttachment){
+        try {
+            $this->course_db->query("INSERT INTO posts (thread_id, parent_id, author_user_id, content, timestamp, anonymous, deleted, endorsed_by, resolved, type, has_attachment) VALUES (?, ?, ?, ?, current_timestamp, ?, ?, ?, ?, ?, ?)", array($thread_id, $parent_post, $user, $content, $anonymous, 0, NULL, 0, $type, true));
+            $this->course_db->query("DELETE FROM viewed_responses WHERE thread_id = ?", array($thread_id));
+            //retrieve generated thread_id
+            $this->course_db->query("SELECT MAX(id) as max_id from posts where thread_id=? and author_user_id=?", array($thread_id, $user));
+        } catch (DatabaseException $dbException){
+            if($this->course_db->inTransaction()){
+                $this->course_db->rollback();
+            }
+        }
+    }
+
```

Comment:
This seems to be an almost exact duplicate of createPost is there a reason not to use that function and just modify the parent_id that the post is replying to?

Reply:
I'll modify this bit. I was experimenting with things earlier and I totally forgot about this change